crates/sui-core/src/authority/authority_store.rs

```diff
@@ -621,24 +639,36 @@ impl AuthorityStore {
 
     /// Checks if the input object identified by the InputKey exists, with support for non-system
     /// packages i.e. when version is None.
-    pub fn multi_input_objects_exist(
+    /// If the shared object was previously deleted as indicated by its existence in the object
+    /// marker table, we consider it to be existing so that the transaction manager proceed
+    pub fn multi_input_objects_exists(
         &self,
         keys: impl Iterator<Item = InputKey> + Clone,
+        epoch_id: EpochId,
     ) -> Result<Vec<bool>, SuiError> {
         let (keys_with_version, keys_without_version): (Vec<_>, Vec<_>) =
             keys.enumerate().partition(|(_, key)| key.1.is_some());
 
-        let versioned_results = keys_with_version.iter().map(|(idx, _)| *idx).zip(
+        let mut versioned_results = vec![];
+        for ((idx, input_key), found) in keys_with_version.iter().zip(
             self.perpetual_tables
                 .objects
                 .multi_get(
                     keys_with_version
                         .iter()
                         .map(|(_, k)| ObjectKey(k.0, k.1.unwrap())),
                 )?
-                .into_iter()
-                .map(|o| o.is_some()),
-        );
+                .into_iter(),
+        ) {
+            match found {
+                Some(_) => versioned_results.push((*idx, true)),
+                None => {
+                    let shared_object_deleted =
+                        self.shared_object_deleted(&input_key.0, epoch_id)?;
```

Comment:
we can leave this for now, but this code is pretty performance critical and we will almost certainly need to do a multi-get here. Even better will be if we know whether the InputKey is a shared object which will let us skip this for most objects - please add a TODO